typescript/utils/src/utils.ts

```diff
@@ -347,11 +347,16 @@ export async function runWithTimeout<T>(
   timeoutMs: number,
   callback: () => Promise<T>,
 ): Promise<T | void> {
-  const timeout = new Promise<void>((_, reject) =>
-    setTimeout(
-      () => reject(new Error(`Timed out in ${timeoutMs}ms.`)),
-      timeoutMs,
-    ),
+  let timeout: NodeJS.Timeout;
+  const timeoutProm = new Promise<void>(
+    (_, reject) =>
+      (timeout = setTimeout(
+        () => reject(new Error(`Timed out in ${timeoutMs}ms.`)),
+        timeoutMs,
+      )),
   );
-  return Promise.race([callback(), timeout]);
+  const ret = Promise.race([callback(), timeoutProm]);
```

Comment:
hmm I don't think I understand exactly how this works
I saw @jmrossy has a slightly different implementation... https://github.com/hyperlane-xyz/hyperlane-explorer/blob/1fb99eb0bde189bc1be5b909ef7f9a34bb8f0cf6/src/utils/timeout.ts#L49

Reply:
What we need to be able to do is clear the timeout if the other promise resolves first, I'm not sure his implementation does that.

Reply:
You were probably confused b/c I meant to await Promise.race and did not?